agent/structs/config_entry_intentions.go

```diff
@@ -751,24 +811,37 @@ func (e *ServiceIntentionsConfigEntry) validate(legacyWrite bool) error {
 }
 
 // Wildcard usage verification
-func validateIntentionWildcards(name string, entMeta *EnterpriseMeta) error {
+// position argument is passed to wrapValidationErrorString for better error messages
+func validateIntentionWildcards(name string, entMeta *EnterpriseMeta, position string) error {
```

Comment:
We should avoid allocating/passing args that are only used for error handling; I think the positional information can be wrapped by the caller of this func instead of it being passed down just to format an error message.